packages/langium/src/lsp/semantic-token-provider.ts

```diff
@@ -205,6 +205,8 @@ export abstract class AbstractSemanticTokenProvider implements SemanticTokenProv
         this.currentRange = undefined;
         this.currentDocument = document;
         this.currentTokensBuilder = this.getDocumentTokensBuilder(document);
+        // Flush the current state - we don't want to perform a delta highlighting
+        this.currentTokensBuilder.previousResult(this.currentTokensBuilder.id);
```

Comment:
This line looks cryptic to me, because I have not so much knowledge about Langium. Also the comment isn't really helpful.

Could you add a FLUSH method to the token builder instead? All what is needed comes from the token builder, so I guess it could be a good idea.